spec/models/tenant_quota_spec.rb

```diff
@@ -1,5 +1,116 @@
 require 'spec_helper'
 
 describe TenantQuota do
-  pending "add some examples to (or delete) #{__FILE__}"
+  let(:settings) {{}}
+  let(:tenant)   {Tenant.new(:domain => 'x.com', :parent => default_tenant)}
+
+  let(:default_tenant) do
+    Tenant.seed
+    Tenant.default_tenant
+  end
+
+  let(:root_tenant) do
+    Tenant.seed
+    Tenant.root_tenant
+  end
+
+  context "validations" do
+    it "rejects invalid name" do
+      expect { described_class.create!(:name => "XXX") }.to raise_error
+    end
+
+    it "rejects missing value" do
+      expect { described_class.create!(:name => :cpu_allocated, :unit => "mhz") }.to raise_error
+    end
+
+    it "rejects missing unit" do
+      expect { described_class.create!(:name => :cpu_allocated, :value => 4096) }.to raise_error
+    end
+  end
+
+  describe ".set" do
+    it "adds new quotas" do
+      described_class.set(default_tenant, :cpu_allocated => {:value => 1024}, :vms_allocated => {:value => 20}, :mem_allocated => {:value => 4096})
```

Comment:
@jrafanie By doing it this way we are actually hiding the details of the internal structure of quotas from the UI. The UI doesn't need to know it's even talking to ActiveRecord. This way similar calls can eventually be made through a REST API without knowing the backend implementation and the same generic structures can be passed back and forth.